sr_data_visualization/src/sr_data_visualization/tab_options.py

```diff
@@ -0,0 +1,259 @@
+#!/usr/bin/env python3
+
+# Copyright 2022 Shadow Robot Company Ltd.
+#
+# This program is free software: you can redistribute it and/or modify it
+# under the terms of the GNU General Public License as published by the Free
+# Software Foundation version 2 of the License.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+# FITNESS FOR A PARTICULAR PURPOSE. See the GNU General Public License for
+# more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program. If not, see <http://www.gnu.org/licenses/>.
+
+from __future__ import absolute_import
+
+import os
+import rospkg
+
+from python_qt_binding.QtGui import QIcon
+
+from python_qt_binding.QtWidgets import (
+    QPushButton,
+    QWidget,
+    QGridLayout,
+    QRadioButton,
+    QHBoxLayout,
+    QGroupBox,
+)
+
+
+class GenericTabOptions(QWidget):
+    def __init__(self, tab_name, parent=None):
+        super().__init__(parent=parent)
+
+        ICON_DIR = os.path.join(
+            rospkg.RosPack().get_path('sr_visualization_icons'), 'icons')
+        self.ICONS = {
+            'GREEN': QIcon(os.path.join(ICON_DIR, 'green.png')),
+            'RED': QIcon(os.path.join(ICON_DIR, 'red.png')),
+            'BLUE': QIcon(os.path.join(ICON_DIR, 'blue.png')),
+            'MAGENTA': QIcon(os.path.join(ICON_DIR, 'magenta.png')),
+            'YELLOW': QIcon(os.path.join(ICON_DIR, 'yellow.png')),
+            'CYAN': QIcon(os.path.join(ICON_DIR, 'cyan.png'))
+        }
+
+        self.init_ui()
+        self.create_tab_options()
+        self.setLayout(self.layout)
+
+    def init_ui(self):
+        self.layout = QGridLayout(self)
+        self.layout.setContentsMargins(0, 0, 0, 0)
+
+    def create_tab_options(self):
+        groupbox = QGroupBox("Graph Options")
+        self.layout.addWidget(groupbox)
+
+        self.check_layout = QHBoxLayout()
+        groupbox.setLayout(self.check_layout)
+
+        self.create_variable_trace_buttons()
+        self.create_common_buttons()
+
+    def create_variable_trace_buttons(self):
```

Comment:
why not simply create_trace_buttons?

Reply:
because the "All" radio button is in the function below which is also a trace button